main.js

```diff
@@ -0,0 +1,165 @@
+
+var setDefaultLocation = function() {
+  //if zipcode, store the zipcode
+  //if no zipcode, store the lat/long
+  var defaultValues = {};
+  if (searchLocation.zip){
+    defaultValues.zip = searchLocation.zip;
+  } else {
+    defaultValues.latitude = searchLocation.latitude;
+    defaultValues.longitude = searchLocation.longitude;
+  }
+  var defaultLocation = JSON.stringify(defaultValues);
+  localStorage.defaultLocation = defaultLocation;
+}
+
+var getDefaultLocation = function() {
+  return localStorage.defaultLocation ? JSON.parse(localStorage.defaultLocation) : {};
+}
+
+var searchLocation = getDefaultLocation();
+
+var weatherConditions = {
+};
+
+var forecastData = {
+  forecasts: []
+};
+
+var weatherViewTemplate = Handlebars.compile($('#weather-view-template').html());
+var mapTemplate = Handlebars.compile($('#map-template').html());
+var forecastViewTemplate = Handlebars.compile($('#forecast-view-template').html());
+
+var getLocation = function() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      searchLocation.latitude = Math.round(position.coords.latitude);
+      searchLocation.longitude = Math.round(position.coords.longitude);
+    });
+  }
+};
+
+var setQueryString = function() {
+  var queryString = '';
+  if (searchLocation.zip) {
+    queryString = `zip=${searchLocation.zip}`
+  } else {
+    queryString = `lat=${searchLocation.latitude}&lon=${searchLocation.longitude}`;
+  }
+  return queryString;
+}
+
+var setGoogleString = function(){
+  var googleObj = {};
+  if (searchLocation.zip){
+    googleObj.googleQuery = searchLocation.zip;
+  } else {
+    googleObj.googleQuery = `${searchLocation.latitude}%2C${searchLocation.longitude}`;
+  }
+  return googleObj;
+}
+
+var setConditions = function(weatherObj){
```

Comment:
Again with the set/get issue, but also, be more descriptive.. get conditions for what?